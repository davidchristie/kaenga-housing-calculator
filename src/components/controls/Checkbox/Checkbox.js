import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Checkbox as BootstrapCheckbox } from 'react-bootstrap'

class Checkbox extends Component {
  render () {
    const {
      disableIfUnchecked,
      inline,
      input: {
        onChange,
        value
      },
      label
    } = this.props
    return (
      <BootstrapCheckbox
        checked={value}
        disabled={disableIfUnchecked && !value}
        inline={inline}
        onChange={onChange}
      >
        {label}
      </BootstrapCheckbox>
    )
  }
}

Checkbox.defaultProps = {
  input: {}
}

Checkbox.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.any
  }).isRequired,
  label: PropTypes.string
}

export default Checkbox
