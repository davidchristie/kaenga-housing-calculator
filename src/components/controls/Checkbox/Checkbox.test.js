import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'

import Checkbox from './Checkbox'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Checkbox />)).toMatchSnapshot()
})
