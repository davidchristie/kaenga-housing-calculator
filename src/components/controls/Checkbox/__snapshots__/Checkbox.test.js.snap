exports[`test matches snapshot 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={undefined}
  disabled={false}
  inline={false}
  onChange={undefined} />
`;
